src/Raven.Server/Documents/Indexes/Test/TestIndexItemEnumerator.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Indexes.Test;
+
+public class TestIndexItemEnumerator : IIndexedItemEnumerator
+{
+    private readonly IIndexedItemEnumerator _inner;
+    private int _count;
+    private readonly int _max;
+
+    public TestIndexItemEnumerator(IIndexedItemEnumerator inner, int collections, int maxDocumentsPerIndex)
+    {
+        _max = maxDocumentsPerIndex / collections;
```

Comment:
`_max` => `_maxOutputsPerCollection`?
More informative than `_max`. :)

Reply:
Fixed, it's more meaningful now

Reply:
For the configuration option idea, I don't think anyone would want stale results in case of test index. I added optional parameter of `WaitForNonStaleResultsTimeout` though, if default 15 seconds is not enough.